src/modules/map/components/toolbox/TimeSlider.vue

```diff
@@ -49,12 +49,56 @@ const layersWithTimestamps = computed(() =>
 )
 const activeLayers = computed(() => store.state.layers.activeLayers)
 const previewYear = computed(() => store.state.layers.previewYear)
-const invalidYear = computed(
-    () =>
-        displayedYear.value != previewYear.value &&
-        displayedYear.value.toString().length == 4 &&
-        !allYears.value.includes(parseInt(displayedYear.value))
-)
+
+const isInputYearValid = ref(true)
+
+/**
+ * Used for the year in the input field Validate the input from the user. In case it's invalid, we
+ * don't propagate the value to the state, but instead save it to an intermediate state variable to
+ * be displayed along with an error message. This is important so that the cursor doesn't jump
+ * around to invalid years
+ */
+const inputYear = computed({
+    get() {
+        return falseYear.value || currentYear.value
+    },
+    set(value) {
+        if (value.length < 4) return value
+
+        value = parseInt(value)
+
+        // only if the year is valid we write this to the property
+        // otherwise we show errors
+        if (!allYears.value.includes(value)) {
+            isInputYearValid.value = false
+            falseYear.value = value
+        } else {
+            isInputYearValid.value = true
+            currentYear.value = parseInt(value)
+            falseYear.value = null
+        }
+    },
+})
+
+/**
+ * The currently selected year
+ *
+ * We can't use the store value directly, as sometimes we want to update the current year displayed
+ * without updating the store too much
+ */
+const currentYear = computed({
+    get() {
+        return _currentYear.value
+    },
+    set(value) {
+        _currentYear.value = value
+
+        // we're currently dragging the cursor, don't update the store
+        if (yearCursorIsGrabbed) return
+
+        dispatchCurrentYearToStore()
+    },
```

Comment:
For the discussion about early return, I think here is a good example of where the early returns make the function harder to read as you have a return in a middle and having such 

```javascript
    set(value) {
        _currentYear.value = value
        // we're currently dragging the cursor, don't update the store
        if (!yearCursorIsGrabbed) {
            dispatchCurrentYearToStore()
        }
    },
```
Especially here you don't have any issue with indentation because you have everything in a function call.
Moreover always use `if (...)` with brackets

Reply:
Hmm, definitely true :+1: . I think this is a leftover from where there was more in the function before I changed stuff. Now it doesn't make sense anymore.